Scripts/Proxies/TwoHandedProxyBase.cs

```diff
@@ -277,14 +339,83 @@ void RemoveFeedbackRequest(ProxyFeedbackRequest request)
 
         public void ClearFeedbackRequests(IRequestFeedback caller)
         {
-            var requests = caller == null
-                ? new List<ProxyFeedbackRequest>(m_FeedbackRequests)
-                : m_FeedbackRequests.Where(feedbackRequest => feedbackRequest.caller == caller).ToList();
+            // Interate over keys instead of pairs in the dictionary, in order to prevent out-of-sync errors when exiting EXR
+            foreach (var key in m_FeedbackRequests.Keys.ToList())
+            {
+                if (key != null && key.caller == caller)
+                    RemoveFeedbackRequest(key);
+            }
+        }
+
+        void UpdateVisibility()
+        {
+            var rightProxyRequestsExist = false;
+            var leftProxyRequestsExist = false;
+            var shakenVisibility = m_SemitransparentLockRequest != null;
+            if (shakenVisibility)
+            {
+                // Left & right device affordances should be visible when the input-device is shaken
+                rightProxyRequestsExist = true;
+                leftProxyRequestsExist = true;
+            }
+            else if (m_FeedbackRequests.Count > 0)
+            {
+                // Find any visible feedback requests for each hand
+                foreach (var request in m_FeedbackRequests)
+                {
+                    var node = request.Key.node;
+                    var visible = request.Key.visible;
+
+                    if (!leftProxyRequestsExist)
+                        leftProxyRequestsExist = node == Node.LeftHand && visible;
+
+                    if (!rightProxyRequestsExist)
+                        rightProxyRequestsExist = node == Node.RightHand && visible;
+
+                    if (rightProxyRequestsExist && leftProxyRequestsExist)
+                        break;
+                }
+            }
+
+            rightAffordanceRenderersVisible = rightProxyRequestsExist;
+            leftAffordanceRenderersVisible = leftProxyRequestsExist;
+
+            rightBodyRenderersVisible = shakenVisibility;
+            leftBodyRenderersVisible = shakenVisibility;
+        }
+
+        IEnumerator MonitorFeedbackRequestLifespan(ProxyFeedbackRequest request)
```

Comment:
On second review, I'm not sure I like this coupled to all feedback definitions. We may find that there are more situations where we want to monitor feedback, but for now I would rather this was just a single coroutine that we start when detect a shake that adds the feedback (if not already added), waits a period of time, and then removes it. This avoids turning the feedback list into a dictionary.

Reply:
As per our slack discussion, this covers all feedback countdown durations, not just shake.  So leaving in-place for now, as this is the means of ticking display duration for each request (checked against in UpdateVisibility for each hand).